### PR TITLE
feat) add support for the `S1` SKU

### PR DESCRIPTION
**Pull request by SudhindraKovalam**
## Checklist

- [ x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only) (Since the change is simple, I was told, no design review needed)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by cormacpayne** [comment]
@refortie I re-targeted this PR at the new default branch, `master`, so you'll need to pull from that branch rather than `preview` to resolve the above merge conflicts whenever you are able to get back to this PR.

_Note_: since the next release is the GA of the `Az` module, we have removed all reference docs to `AzureRM` cmdlets, so you will need to regenerate your help based on the `Az` cmdlets.

File: src/ResourceManager/Maps/Commands.Maps/MapsAccount/NewAzureMapsAccount.cs [hunk 1]
```diff
@@ -52,7 +52,7 @@ public class NewAzureMapsAccount : MapsAccountBaseCmdlet
             Mandatory = true,
             ValueFromPipelineByPropertyName = true,
             HelpMessage = "Maps Account Sku Name.")]
-        [ValidateSet("S0")]
+        [ValidateSet("S0", "S1")]
```
Comment (markcowl) [inline src/ResourceManager/Maps/Commands.Maps/MapsAccount/NewAzureMapsAccount.cs]: We should move away from using ValidateSet to using PSArgumentCompleter - this allows users to tab through known vaues, but enter values outside the known values list.

Also, longer term, is this set of values somethign we should discover from API?

Comment (SudhindraKovalam) [reply src/ResourceManager/Maps/Commands.Maps/MapsAccount/NewAzureMapsAccount.cs]: @markcowl, made the change to use `PSArugementCompleter`

File: src/ResourceManager/Maps/Maps.Test/ScenarioTests/MapsAccountTests.ps1 [hunk 2]
```diff
@@ -49,6 +49,43 @@ function Test-NewAzureRmMapsAccount
     }
 }
 
+<#
+.SYNOPSIS
+Test New-AzureRmMapsAccountS1
+#>
+function Test-NewAzureRmMapsAccountS1
```
Comment (cormacpayne) [inline src/ResourceManager/Maps/Maps.Test/ScenarioTests/MapsAccountTests.ps1]: @SudhindraKovalam this test isn't currently being run as a part of CI -- you'll need to create a new test method that calls this function in the [`MapsAccountTests.cs`](https://github.com/Azure/azure-powershell/blob/master/src/ResourceManager/Maps/Maps.Test/ScenarioTests/MapsAccountTests.cs) file. Once added, you'll need to add the corresponding `.json` file containing the recording for this test that will be used as a part of playback during CI

File: src/ResourceManager/Maps/Commands.Maps/AzureRM.Maps.psd1 [hunk 3]
```diff
@@ -12,7 +12,7 @@
 # RootModule = ''
 
 # Version number of this module.
-ModuleVersion = '1.0.5'
+ModuleVersion = '1.0.6'
```
Comment (cormacpayne) [inline src/ResourceManager/Maps/Commands.Maps/AzureRM.Maps.psd1]: @SudhindraKovalam you can revert the change in this file -- this file is going to be deleted in favor of `Az.Maps.psd1` in the next week

**Comment by markcowl** [comment]
@azuresdkci add to whitelist

**Comment by markcowl** [comment]
@azuresdkci test this please

File: src/ResourceManager/Maps/Maps.Test/ScenarioTests/MapsAccountTests.ps1 (hunk 2 above)
Comment (cormacpayne) [reply src/ResourceManager/Maps/Maps.Test/ScenarioTests/MapsAccountTests.ps1]: Update: we will enable this test once the work to restructure the repository has been completed.
